chain/network/src/types.rs

```diff
@@ -345,15 +348,58 @@ pub enum NetworkClientResponses {
     Ban { ban_reason: ReasonForBan },
 }
 
-/// Adapter to break dependency of sub-components on the network requests.
-/// For tests use `MockNetworkAdapter` that accumulates the requests to network.
-pub trait PeerManagerAdapter: Sync + Send {
-    fn send(
-        &self,
-        msg: PeerManagerMessageRequest,
-    ) -> BoxFuture<'static, Result<PeerManagerMessageResponse, actix::MailboxError>>;
+pub trait MsgRecipient<M: actix::Message>: Send + Sync + 'static {
```

Comment:
This seems to be a change to the scaffolding infra, which is more or less unrelated to the behavioral/feature change here.

Reply:
It is related, because without it I cannot make PeerManagerActor a handler for a new message type. It is just an adjustment of tests after implementing a feature. Anyway, extracted to a separate PR.